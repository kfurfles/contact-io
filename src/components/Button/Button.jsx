import React from 'react'

import './style.scss'

const Button = ({ selected, direction, children, ...props }) => {
    const flow = selected && direction === 'up' ? 'fas fa-sort-up' : 'fas fa-sort-down'

    const selectedClassName = selected ? 'is-selected' : ''
    const flowClassName = direction === 'up' && selected ? 'is-up' : 'is-down'

    const computedClass = `${selectedClassName} ${flowClassName}`

    return (
        <button className={`ct-button ${computedClass}`.trim()} {...props}>
            {children} <i className={flow} />
        </button>  
    )
}

Button.defaultProps = {
    selected: false,
    direction: true
}

export default Button