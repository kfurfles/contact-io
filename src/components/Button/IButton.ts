import { ButtonHTMLAttributes } from "react";

export interface IButton extends ButtonHTMLAttributes<HTMLButtonElement> {
    selected: boolean,
    direction: 'up' | 'down'
}