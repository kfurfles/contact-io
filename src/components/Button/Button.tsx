import React from 'react'
import {IButton } from './IButton'

import './style.scss'

const Button: React.FC<IButton> = ({ selected, direction, children, ...props }) => {
    const flow = selected && direction === 'up' ? 'fas fa-sort-up' : 'fas fa-sort-down'

    const selectedClassName = selected ? 'is-selected' : ''
    const flowClassName = direction === 'up' && selected ? 'is-up' : 'is-down'

    const computedClass = `${selectedClassName} ${flowClassName}`

    return (
        <button className={`ct-button ${computedClass}`.trim()} {...props}>
            {children} <i className={flow} />
        </button>  
    )
}

Button.defaultProps = {
    selected: false,
    direction: 'down'
}

export default Button