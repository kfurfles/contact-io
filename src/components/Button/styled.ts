import styled from '@emotion/styled'
import { IButton } from './IButton'

export const Button = styled.button<IButton>`
    align-items: center;
    background-color: ${props => props.selected ? '#4C84FF' : '#fff' };
    border-radius: 5px;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 2px 2px rgba(0, 0, 0, .07);
    color: ${props => props.selected ? '#fff' : '#9FA9BC' };
    display: flex;
    flex-wrap: nowrap;
    font-size: 1.3rem;
    font-weight: normal;
    justify-content: center;
    margin: 0 8px;
    padding: 10px;
    white-space: nowrap;

    i {
        color: ${props => props.selected ? '#fff' : '#9FA9BC' };
        margin-top: ${props => props.direction === 'up' ? '5px' : '-5px' };
        margin-left: 8px;
    }
`