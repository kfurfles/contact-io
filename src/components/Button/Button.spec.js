import React from 'react';
import { render} from '@testing-library/react';

import Button from './index';

const mockButtonProps = {
    selected: true,

};

describe('Button', () => {
    it('Should render Button', () => {
        const { fragment } = render(<Button/>);
        expect(fragment).toMatchSnapshot();
    });

    it('Should render selected button', () =>{
        const fragment = render(<Button selected={true}>text-button</Button>);

        const classIndex = fragment.getByText('text-button').className.indexOf('is-selected')
        expect(classIndex > -1).toBe(true)
    })

    it('Should render selected with ascend options order button', () =>{
        const fragment = render(<Button selected={true}  direction={'up'}>text-button</Button>);

        const upIndex = fragment.getByText('text-button').className.indexOf('is-up')
        expect(upIndex > -1).toBe(true)
    })

    it('Should render selected with ascend options order button', () =>{
        const fragment = render(<Button selected={true}  direction={'down'}>text-button</Button>);

        const downIndex = fragment.getByText('text-button').className.indexOf('is-down')
        expect(downIndex > -1).toBe(true)
    })
});