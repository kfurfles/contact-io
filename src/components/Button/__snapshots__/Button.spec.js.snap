// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render Button 1`] = `undefined`;
